BGL/include/CGAL/boost/graph/io.h

```diff
@@ -401,6 +402,284 @@ bool write_inp(std::ostream& os,
+    }
+  write_vector<FT>(os,coordinates);
+}
+
+  } // end namespace CGAL::internal::write_vtp
+} // end namespace CGAL::internal
+
+/*!\ingroup PkgBGLIOFct
+ *
+ * \brief  writes a triangulated surface mesh in the `PolyData` XML format.
+ *
+ * \tparam TriangleMesh a model of `FaceListGraph` with only triangle faces.
+ * \tparam NamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+ *
+ * \param os the stream used for writting.
+ * \param mesh the triangle mesh to be written.
+ * \param mode decides if the data should be written in binary (`IO::BINARY`)
+ *   or in ASCII (`IO::ASCII`). `IO::BINARY` is the default.
+ * \param np optional sequence of \ref pmp_namedparameters "Named Parameters" among the
+ * ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *    \cgalParamBegin{vertex_point_map} the property map with the points associated to
+ * the vertices of `mesh`. If this parameter is omitted, an internal property map for
+ *       `CGAL::vertex_point_t` must be available in `TriangleMesh`.
+ *     \cgalParamEnd
+ *    \cgalParamBegin{vertex_index_map} the property map with the indices associated to
+ * the vertices of `mesh`. If this parameter is omitted, an internal property map for
+ *       `CGAL::vertex_index_t` must be available in `TriangleMesh`.
+ *     \cgalParamEnd
+ * \cgalNamedParamsEnd
+ */
+template<class TriangleMesh,
+         class NamedParameters>
+void write_vtp(std::ostream& os,
+               const TriangleMesh& mesh,
+               IO::Mode mode,
+               const NamedParameters& np)
+{
+  os << "<?xml version=\"1.0\"?>\n"
+     << "<VTKFile type=\"PolyData\" version=\"0.1\"";
+#ifdef CGAL_LITTLE_ENDIAN
+  os << " byte_order=\"LittleEndian\"";
+#else // CGAL_BIG_ENDIAN
+  os << " byte_order=\"BigEndian\"";
+#endif
+  switch(sizeof(std::size_t)) {
+  case 4: os << " header_type=\"UInt32\""; break;
+  case 8: os << " header_type=\"UInt64\""; break;
+  default: CGAL_error_msg("Unknown size of std::size_t");
+  }
+  os << ">\n"
+     << "  <PolyData>" << "\n";
+
+  os << "  <Piece NumberOfPoints=\"" << num_vertices(mesh)
+     << "\" NumberOfPolys=\"" << num_faces(mesh) << "\">\n";
+  std::size_t offset = 0;
+  const bool binary = (mode == IO::BINARY);
+  internal::write_vtp::write_points_tag(os,mesh,binary,offset, np);
+  internal::write_vtp::write_polys_tag(os,mesh,binary,offset, np);
+  os << "   </Piece>\n"
+     << "  </PolyData>\n";
+  if (binary) {
+    os << "<AppendedData encoding=\"raw\">\n_";
+    internal::write_vtp::write_polys_points(os,mesh, np);
+    internal::write_vtp::write_polys(os,mesh, np);
+  }
+  os << "</VTKFile>\n";
+}
+
+/*! \ingroup PkgBGLIOFct
+ *  Overload that implements the default arguments
+ *
+ *  Calls `write_vtp(os, mesh, mode, parameters::all_default())`.
+ */
+template<class TriangleMesh>
+void write_vtp(std::ostream& os,
+                      const TriangleMesh& mesh,
+               IO::Mode mode = IO::BINARY)
```

Comment:
actually mode should be a named parameter. I can do it's simpler for you. We just need a name for the named parameter (`use_binary_mode(false/true)`?)

Reply:
Yes, you are right. That would be a simpler API. `use_binary_mode` seems fine for the name.